js/form_question_selectable.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/* global sortable */
+
+class GlpiFormQuestionTypeSelectable {
+
+    /**
+     * The selectable input type.
+     *
+     * @type {string}
+     */
+    #inputType;
+
+    /**
+     * The options container.
+     *
+     * @type {JQuery<HTMLElement>}
+     */
+    #container;
+
+    /**
+     * Create a new GlpiFormQuestionTypeSelectable instance.
+     *
+     * @param {JQuery<HTMLElement>} container
+     */
+    constructor(inputType = null, container = null) {
+        this.#inputType = inputType;
+        this.#container = $(container);
+
+        if (this.#container !== null) {
+            this.#container.children()
+                .each((index, option) => this.#registerOptionListeners($(option)));
+
+            // Register listeners for the empty option
+            this.#container
+                .siblings('div[data-glpi-form-editor-question-extra-details]')
+                .each((index, option) => this.#registerOptionListeners($(option)));
+
+            // Compute the state to update the input names
+            window.glpi_form_editor_controller.computeState();
+
+            // Restore the checked state
+            if (this.#inputType === 'radio') {
+                this.#container
+                    .find('input[type="radio"][checked]')
+                    .prop('checked', true);
+            }
+        }
+    }
+
+    #registerOptionListeners(option) {
+        option
+            .find('input[type="text"]')
+            .on('input', (event) => this.handleOptionChange(event))
+            .on('keydown', (event) => this.handleKeydown(event));
+
+        option
+            .find('i[data-glpi-form-editor-question-option-remove]')
+            .on('click', (event) => this.removeOption(event));
+    }
+
+    enableOptionsSortable() {
```

Comment:
Maybe it should be called by the contructor ?
The only two times this function is called is right after an item was created.

This would allow the method to be protected.